component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/utils/TokenExchangeUtils.java

```diff
@@ -752,9 +752,11 @@ public static boolean checkExpirationTime(Date expirationTime, long currentTimeI
 
         long expirationTimeInMillis = expirationTime.getTime();
         if ((currentTimeInMillis + timeStampSkewMillis) > expirationTimeInMillis) {
-            handleException(OAuth2ErrorCodes.INVALID_REQUEST, "JSON Web Token is expired." + ", Expiration Time(ms) "
+            String errorMessage = "JSON Web Token is expired." + ", Expiration Time(ms) "
                     + ":" + " " + expirationTimeInMillis + ", TimeStamp Skew : " + timeStampSkewMillis + ", Current "
-                    + "Time : " + currentTimeInMillis + ". JWT Rejected and validation terminated");
+                    + "Time : " + currentTimeInMillis + ". JWT Rejected and validation terminated";
+            log.debug(errorMessage);
+            throw new IdentityOAuth2Exception(OAuth2ErrorCodes.INVALID_REQUEST, errorMessage);
```

Comment:
The best solution would be creating a separate method like `handleException` but for client exceptions.